apps/sms/views/shared/js/navigation.js

```diff
@@ -1,361 +1,920 @@
+  () => {};
+
+/**
+  * This contains an array of view characteristics, necessary to handle
+  * navigation requests.
+  * Each item has the following properties:
+  * - name: this is the name of the view. This will be used also in the URL
+  *   hash if the same HTML file holds several views.
+  * - url: this is the URL used to match the current location and find the
+  *   current view.
+  * - behavior: this is the actual JavaScript object that manages this view.
+  * - previous (optional): this is the name of the previous view; this is
+  *   necessary to decide if we do a forward/backward animation.
+  * - partOf (optional): this is used when a view is part of another view's
+  *   HTML document. In that case one view is "master" and the other are only
+  *   part of the master view.
+  *
+  * NGA_VIEWS holds the configuration for NGA split views.
+  */
+const NGA_VIEWS = Object.freeze([
+  Object.freeze({
+    name: 'thread',
+    url: 'views/conversation/index.html',
+    behavior: 'ConversationView',
+    previous: 'thread-list'
+  }),
+  Object.freeze({
+    name: 'thread-list',
+    url: 'views/inbox/index.html',
+    behavior: 'InboxView'
+  }),
+  Object.freeze({
+    name: 'report-view',
+    behavior: 'ReportView',
+    previous: 'thread',
+    partOf: 'thread'
+  }),
+  Object.freeze({
+    name: 'composer',
+    behavior: 'ConversationView',
+    previous: 'thread-list',
+    partOf: 'thread'
+  }),
+  Object.freeze({
+    name: 'group-view',
+    behavior: 'GroupView',
+    previous: 'thread',
+    partOf: 'thread'
+  })
+]);
+
+/**
+  * OA_VIEWS holds the configuration for the old non-split architecture.
+  */
+const OA_VIEWS = Object.freeze([
+  Object.freeze({
+    name: 'thread-list',
+    url: 'index.html',
+    behavior: 'InboxView'
+  }),
+  Object.freeze({
+    name: 'thread',
+    behavior: 'ConversationView',
+    previous: 'thread-list',
+    partOf: 'thread-list'
+  }),
+  Object.freeze({
+    name: 'composer',
+    behavior: 'ConversationView',
+    previous: 'thread-list',
+    partOf: 'thread-list'
+  }),
+  Object.freeze({
+    name: 'group-view',
+    behavior: 'GroupView',
+    previous: 'thread',
+    partOf: 'thread-list'
+  }),
+  Object.freeze({
+    name: 'report-view',
+    behavior: 'ReportView',
+    previous: 'thread',
+    partOf: 'thread-list'
+  })
+]);
+
+const VIEWS = isUsingOldArchitecture() ? OA_VIEWS : NGA_VIEWS;
+
+var currentState;
+var navigationTransition;
+
+/**
+ * Navigate to this URL.
+ *
+ * @param {String} url URL to load.
+ */
+function setLocation(url) {
+  debug('Setting location', url);
+  window.location.assign(url);
+}
 
-// This function is used as a rejection handler, to swallow errors and log them
-function catchError(e) {
-  if (e) {
+/**
+ * We're using the old architecture if we're not in a split document.
+ *
+ * @returns {Boolean} true if we're using the old architecture.
+ */
+function isUsingOldArchitecture() {
```

Comment:
__question:__ Just a trivial question: Will we need this method for smth like our clients? It might be good if we can provide this in view utils, but it seems we only this temporary.